### PR TITLE
Update documentation of composite provider flag.

### LINKED ISSUE #6173: Improve documentation for CompositeProviderFlag
The javadoc https://helidon.io/docs/v2/apidocs/io.helidon.security/io/helidon/security/CompositeProviderFlag offers very little guidance in helping users choose the appropriate configuration when composite providers are configured.

One recent question from slack has the user configuring two providers: (1) OID, with a fallback of (@) Basic-Auth. The user was configuring both providers as "MAY_FAIL". However, the appropriate configuration should be having both configured as "SUFFICIENT" with optional flag set to true.

This configuration, however, is entirely not obvious from reading the documentation. We should therefore improve the documentation - probably by providing examples for when/how it is appropriate for using compositions that include MAY_FAIL, SUFFICIENT, and optional=true - as well as the other flags available.

Also related to this is considering how the OID provider was not available due to a network outage (or perhaps a 404). We should also mention this case in particular to indicate something like "a failure to reach the security provider" will be treated as a failure, etc.

bad:

```
security:
  enabled: true
  provider-policy:
    type: "COMPOSITE"
    name: "composite-policy"
    default: true
    authentication:
      - name: "basic-authenticator"
        flag: "MAY_FAIL"
      - name: "oidc-authenticator"
        flag: "MAY_FAIL"
  providers:
  - name: "basic-authenticator"
    http-basic-auth:
      realm: "helidon"
      users:
      - login: "uimuser"
        password: "changeit"
  - name: "oidc-authenticator"
    oidc:
      identity-uri: "http://ohs1.oamdomain.org:30305"
      base-scopes: "FedRserver.Info"
      client-id: topologyClient
      client-secret: Welcome1
      token-endpoint-auth: CLIENT_SECRET_POST
      cookie-name: "OIDC_SESSION"
      cookie-same-site: "Lax"
      header-use: true
      audience: "FedRserver"
      redirect: true
      redirect-uri: "/sr/test/sol005Adapter"
```

Good (partially; needs optional: true for oidc):
```
security:
  enabled: true
  provider-policy:
    type: "COMPOSITE"
    name: "composite-policy"
    default: true
    authentication:
      - name: "basic-authenticator"
        flag: "SUFFICIENT"
      - name: "oidc-authenticator"
        flag: "SUFFICIENT"
```

### PR DESCRIPTION
**Pull request by tomas-langer**
Resolves #6173

### CODE REVIEW

File: security/security/src/main/java/io/helidon/security/CompositeProviderFlag.java [hunk 1]
```diff
@@ -27,12 +27,23 @@
 public enum CompositeProviderFlag {
     /**
      * Provider may fail, nothing happens if it does.
+     * <p>
+     * If succeeds, continues to the next provider.
+     * To create a provider that may fail, yet that is sufficient,
+     * please configure the flag to be {@link #SUFFICIENT}, and set the provider itself ot be optional
```
Comment (ljnelson) [inline security/security/src/main/java/io/helidon/security/CompositeProviderFlag.java:33]: typo

File: security/security/src/main/java/io/helidon/security/CompositeProviderFlag.java [hunk 2]
```diff
@@ -27,12 +27,23 @@
...
+     * when user cannot be authenticated using it.
      */
     MAY_FAIL(EnumSet.of(SecurityResponse.SecurityStatus.SUCCESS,
                         SecurityResponse.SecurityStatus.ABSTAIN,
                         SecurityResponse.SecurityStatus.FAILURE)),
     /**
      * If succeeds, no further providers are called, otherwise same as optional.
+     * <p>
+     * To create a provider that may fail, yet that is sufficient,
+     * use this flag, and set the provider itself ot be optional
```
Comment (ljnelson) [inline security/security/src/main/java/io/helidon/security/CompositeProviderFlag.java:44]: Typo

File: security/security/src/main/java/io/helidon/security/CompositeProviderFlag.java [hunk 3]
```diff
@@ -27,12 +27,23 @@
...
+     * (most Helidon providers support {@code optional} configuration option, so the provider abstains instead of fails
```
Comment (ljnelson) [inline security/security/src/main/java/io/helidon/security/CompositeProviderFlag.java:45]: Unclosed parenthesis

Comment (tomas-langer) [reply security/security/src/main/java/io/helidon/security/CompositeProviderFlag.java]: fixed

**Review by Verdent — APPROVED** [review]

**Pull request merged**